src/quest_manager/models.py

```diff
@@ -975,3 +989,142 @@ def get_minutes_to_complete(self):
+    # e.g. ("media", "Media") or ("image", "Image")
+    allowed_file_type = models.CharField(
+        max_length=50,
+        choices=[(key, key.capitalize()) for key in FILE_MIME_TYPES.keys()],
+        default="media",
+        help_text=("The types of files that can be uploaded by a student for this question.")
+    )
+
+    marker_notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, providing additional details about "
+            "what a student's submission should be or how the question should be marked."
+        ),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=(
+            "If this is checked, the student must complete this "
+            "question to complete the quest."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Indicates the order of the question in the quest so that "
+            "questions are displayed to the student in the correct order."
+        ),
+    )
+
+    quest = models.ForeignKey(Quest, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=20, choices=QuestionType.choices, default=QuestionType.SHORT_ANSWER
+    )
+
+    class Meta:
+        ordering = ["ordinal"]
+        # ensure two questions in the same quest cannot have the same ordinal
+        unique_together = ["quest", "ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal])
+        e.g. "Quest 1 (short_answer - 1)" for the first question in the quest with name "Quest 1"
+        """
+        return f"{self.quest.name} ({self.type} - {self.ordinal})"
+
+
+class QuestionSubmission(models.Model):
+    """Used for storing a student's submission for a question.
+
+    This model is used for all types of questions, including multiple choice, short answer, file upload, etc.
+    When a student submits a quest, each Question will have a QuestionSubmission created for it.
+    """
+
+    response_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "A text response to a question. This is what the teacher will "
+            "look at when marking the question."
+        ),
+    )
+    response_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/submission/%Y/%m/%d",
+        help_text=(
+            "A file response to a question. This is what the teacher will "
+            "look at when marking the question."
+        ),
+    )
+
+    # Similar to a quest's submission being deleted, a question's submission should not be deleted if the question is
+    question = models.ForeignKey(Question, on_delete=models.SET_NULL, null=True)
```

Comment:
If we are not deleting the QuestionSubmission (which I think is a good idea), we'll need to make sure we have good tests covering deleted Questions and ensure those situation are handled.

Reply:
I think this has already been done. Not sure what else could be tested. Maybe I could add a test ensuring that the files also aren't deleted or something.

![image](https://github.com/bytedeck/bytedeck/assets/11304586/192596d2-0a83-40f3-a647-9852caa10582)

Reply:
Looks good!